src/krb5/_change_password.pyx

```diff
@@ -0,0 +1,217 @@
+# Support for Microsoft set/change password was added in MIT 1.7
+
+import collections
+import typing
+
+from krb5._exceptions import Krb5Error
+
+from krb5._ccache cimport CCache
+from krb5._context cimport Context
+from krb5._creds cimport Creds
+from krb5._krb5_types cimport *
+from krb5._principal cimport Principal
+
+
+cdef extern from "python_krb5.h":
+    krb5_error_code krb5_set_password(
+        krb5_context context,
+        krb5_creds *creds,
+        const char *newpw,
+        krb5_principal change_password_for,
+        int *result_code,
+        krb5_data *result_code_string,
+        krb5_data *result_string
+    ) nogil
+
+    krb5_error_code krb5_set_password_using_ccache(
+        krb5_context context,
+        krb5_ccache ccache,
+        const char *newpw,
+        krb5_principal change_password_for,
+        int *result_code,
+        krb5_data *result_code_string,
+        krb5_data *result_string
+    ) nogil
+
+    krb5_error_code krb5_change_password(
```

Comment:
Seems like this might not be present on macOS's Heimdal's implementation

```bash
python -c "import ctypes; print(hasattr(ctypes.CDLL('/System/Library/PrivateFrameworks/Heimdal.framework/Heimdal'), 'krb5_change_password'))"
```

We would have to put this in another file that is conditionally imported if that symbol is present. The `krb5_set_password*` APIs seem to be present though.

Reply:
the `krb5_change_password()` appears deprecated in Heimdal, perhaps MacOS follows that strictly. In MIT both the `set_password()` and `change_password()` are wrappers for the same code so I decided to drop the `change_password()` completely and suggest `set_password()` with `change_password_for=None` to change password for the owner.